lib/site.taxonomy.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Module dependencies.
+ */
+import Term from './site.taxonomy.term';
+
+/**
+ * Taxonomy class
+ */
+class Taxonomy {
+	/**
+	 * Create a Taxonomy instance
+	 *
+	 * @param {String} siteId - site id
+	 * @param {String} taxonomy - taxonomy type
+	 * @param {WPCOM} wpcom - wpcom instance
+	 * @return {Null} null
+	 */
+	constructor( siteId, taxonomy, wpcom ) {
+		if ( ! siteId ) {
+			throw new Error( '`siteId` is not correctly defined' );
+		}
+
+		if ( ! taxonomy ) {
+			throw new Error( '`taxonomy` is not correctly defined' );
+		}
+
+		if ( ! ( this instanceof Taxonomy ) ) {
+			return new Taxonomy( siteId, taxonomy, wpcom );
+		}
+
+		this.wpcom = wpcom;
+
+		this._siteId = encodeURIComponent( siteId );
+		this._taxonomy = encodeURIComponent( taxonomy );
+		this.rootPath = `/sites/${this._siteId}/taxonomies/${this._taxonomy}`;
+	}
+
+	/**
+	 * Get a list of Terms for the Taxonomy
+	 *
+	 * @param {Object} [query] - query object
+ 	 * @param {Function} fn - callback function
+ 	 * @return {Function} request handler
+ 	 */
+	termsList( query, fn ) {
+		let termsPath = this.rootPath + '/terms';
```

Comment:
Also to be consistent can we use `const termsPath = `${ this.rootPath }/terms` ?